datalad/utils.py

```diff
@@ -1081,6 +1081,31 @@ def wrapped_func(*args, **kwargs):
         return wrapped_func
 
 
+@optional_args
+def memoize(f):
```

Comment:
Why not use `@functools.lru_cache` rather than adding custom code?

Reply:
ha ha -- because I forgot about that one!  will abandon this bastard child here and RF to use it